baseapp/baseapp.go

```diff
@@ -896,13 +910,20 @@ func (app *BaseApp) Commit() (res abci.ResponseCommit) {
 
 	// Reset the Check state to the latest committed.
 	//
-	// NOTE: safe because Tendermint holds a lock on the mempool for Commit.
-	// Use the header from this latest block.
+	// NOTE: This is safe because Tendermint holds a lock on the mempool for
+	// Commit. Use the header from this latest block.
 	app.setCheckState(header)
 
 	// empty/reset the deliver state
 	app.deliverState = nil
 
+	defer func() {
+		if app.haltHeight > 0 && uint64(header.Height) == app.haltHeight {
+			app.logger.Info("halting node", "height", app.haltHeight)
```

Comment:
Maybe like `"halting node at height %d per configuration, might be time to upgrade your node!"` or something like that.

Reply:
Hmmm, maybe we can update it yeah. But the last part seems unnecessary to me. What about `halting node at height %d per configuration`?